### DISCUSSION TITLE
About Intel versions

### DISCUSSION THREAD

**Discussion by kolchagov** [vrolife/modern_laptop#51] [General]
I own an Intel based Redmi Book pro 15 (TM2111-Redmi_Book_Pro_15_2022),  current BIOS release is RMAAD5B0P0B0B
All the peripherals, except acpi are working fine.
I've already tried to compile it, creating the necessary folders and it fails. I suppose I need to extract some information from bios image.
Could you help me add patch to acpi for this model?

**Comment by vrolife** [comment]
First, dump all ACPI tables to a folder and decompile them:
```
cd /path/to/folder
acpidump -b
iasl -d *.dat
copy dsdt.dsl dsdt_origin.dsl
```
Modify dsdt.dsl as needed. 
Increase the  [OEMRevision](https://uefi.org/htmlspecs/ACPI_Spec_6_4_html/19_ASL_Reference/ACPI_Source_Language_Reference.html?highlight=acpi%20source%20language#definitionblockterm) of DefinitionBlock. Important!
Generate the path.diff
```
diff -Naru dsdt_origin.dsl dsdt.dsl >patch.diff
```
Copy path.diff to folder "TM2111-Redmi_Book_Pro_15_2022/RMAAD5B0P0B0B"

Attention! The installer support dsdt.dsl only. Because linux can not override SSDTs in some case.

**Reply by perfectsense72** [reply]
Hi, I have Intel based Redmi Book Pro 15 (TM2111-Redmi_Book_Pro_15_2022) with latest RMAAD5B0P1010 BIOS.
When I try to run 
`acpidump -b`
I got this error: 
`Table signature (0x4F324824) is invalid 
Could not get ACPI table at index 27, AE_BAD_HEADER`
Could you advice me, what can I do with this?

UPD: despite the error I got the file dsdt.dat in my folder (along with a lot of other *.dat files) and after running 
`iasl -d *.dat` 
I got dsdt.dsl file.
Could I ask you what to do next? I mean how to modify dsdt.dsl?

**Reply by vrolife** [reply]
The patches in this repo are all for AMD laptops, and I regret to say that I am not familiar with the Intel version either

**Reply by perfectsense72** [reply]
Ok, thank you anyway for your contribution.

**Comment by kolchagov** [comment]
Could you help a bit… My bios seems very diffrent from yours (which is normal as it’s intel platform).
My suspend almost works - when I close the lid it suspends, when I open it - resumes. I can do it a couple of times without any problem.
But pressing the power button (also configured as suspend) leaves the computer in “frozen” state and I need to restart it, holding power key.
I also noticed, that leaving laptop suspended for a long time - for example 8 hours also leads to “frozen” state (maybe some kind of deep sleep?).

Probably if you explain what you change with your bios would help me solve this problem.

PS: I’ve dumped and decompiled DSDT table successfully, just need a clue what to search for.

**Comment by vrolife** [comment]
Seems to be the problem with S4/S5. You can find more information in this page https://01.org/blogs/rzhang/2015/best-practice-debug-linux-suspend/hibernate-issues.
You may need to disable graphics sessions to view kernel logs.
